### ISSUE TITLE
[DEPR]: Logistration Pages -> micro-frontend

### ISSUE THREAD

**Issue by waheedahmed** [openedx/public-engineering#81]
### Proposal Date

26 May 2022

### Ticket Acceptance Date

09 June 2022

### Technology Removal Date

Cannot remove until enterprise flows are implemented in micro-frontend

### First Open edX Named Release Without This Functionality

not sure

### Rationale

As part of our re-platforming effort, we are replacing Django-server-side rendered frontend implementations in favor of React-based micro-frontend implementations. This ticket covers the login, register, forgot, and reset password pages (aka Logistration).

### Removal

We will be removing the older implementation of login, register, forgot, and reset password pages:
1. [Logistration](https://github.com/openedx/edx-platform/blob/8528fc07f630f4c1d4f886517726941bba6ae4ec/openedx/core/djangoapps/user_authn/views/login_form.py#L138)
2. [Reset Password](https://github.com/openedx/edx-platform/blob/8528fc07f630f4c1d4f886517726941bba6ae4ec/openedx/core/djangoapps/user_authn/views/password_reset.py#L348)

### Replacement

We are replacing these pages with a new micro-frontend-based implementation: [openedx/frontend-app-authn](https://github.com/openedx/frontend-app-authn)

### Deprecation

As part of our re-platforming effort, we are replacing Django-server-side rendered frontend implementations in favor of React-based micro-frontend implementations. This ticket covers the login, register, forgot, and reset password pages (aka Logistration).

### Additional Info

The enterprise flows are still using this legacy implementation and we cannot remove this until those are migrated to the new micro-frontend.

**Comment by kdmccormick** [comment]
Update:
* We don't believe the MFE is at 100% feature parity with the legacy frontend (cc @michaelwheeler who brought this up).
* We believe the Authn MFE needs some work in order to meet our branding guidelines.
* It's not been added to a named release yet. We'd like it have the MFE available in a release before deprecating the legacy view.

**Comment by michaelwheeler** [comment]
Regarding the first point mentioned by @kdmccormick, I was thinking specifically of the ability to [add custom fields to the registration page](https://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/configuration/customize_registration_page.html).

It may also be worth checking to ensure that existing auth-related [events and filters](https://github.com/openedx/edx-platform/blob/master/docs/guides/hooks/index.rst) are supported in a robust manner, so that existing extensions relying on those mechanisms don't break.

**Comment by robrap** [comment]
[inform] This DEPR is in the Draft state. It can't move forward without a DEPR coordinator to help update it to match the updated DEPR template. No action is required, unless anyone wants to move this out of Draft.

**Comment by dianakhuang** [comment]
We still need to investigate whether the custom fields have been added to thein the intervening time. @michaelwheeler do you know whether the auth MFE meets your needs at this time?

**Comment by kdmccormick** [comment]
Reference for custom registration fields (the legacy version, I believe): https://docs.openedx.org/en/latest/site_ops/install_configure_run_guide/configuration/customize_registration_page.html

**Comment by michaelwheeler** [comment]
Hey @dianakhuang we are no longer using Open edX.

**Comment by kdmccormick** [comment]
The existence of this PR implies to me that Custom Registration Field have been added to the MFE, but that the documentation is not yet complete:
* https://github.com/openedx/frontend-app-authn/pull/1595

**Comment by feanil** [comment]
Picking this back up as part of the MFE-vs-legacy inventory (openedx/openedx-platform#38936).  I audited the legacy logistration experience against `frontend-app-authn` to figure out what actually blocks this removal, and it comes down to one thing: enterprise.

Where things stand:

- **At parity in the MFE:** base login/register/forgot/reset, dynamic custom registration fields, third-party/social auth, the institution/SSO chooser, and hinted login.  The custom-registration-fields concern that used to hold this up has landed.
- **The remaining gap is enterprise.**  The legacy page renders an enterprise-branded logistration experience (branded sidebar, generic-provider suppression, consent-gated SSO registration, branded errors, field handling) that the MFE doesn't.  The platform knows it - `openedx/core/djangoapps/user_authn/views/login_form.py:176-190` deliberately keeps SAML/TPA (including enterprise SSO) users on the legacy page, with a `TODO` to drop that guard "as soon as the AuthN MFE is fully-featured."  Until the MFE renders these flows, that guard - and the legacy page behind it - has to stay.

The backend groundwork for this is already done: enterprise logistration logic was decoupled into `openedx-filters` pipeline steps so it stays pluggable rather than built into the platform (openedx/openedx-filters#337, openedx/edx-enterprise#2551, openedx/openedx-platform#38105), including an `enterpriseBranding` payload seam for the authn MFE.  The frontend counterpart doesn't exist yet, and because enterprise is meant to be pluggable, the right MFE shape (likely a plugin into authn slots rather than built-in code) is still an open question.

I've opened **openedx/frontend-app-authn#1691** to track that MFE work.  That's the concrete blocker for dropping the legacy login/registration UI - once it's done and the `has_external_provider` guard can be removed, this DEPR is unblocked.
